Client/src/components/maze.tsx

```diff
@@ -31,56 +42,84 @@ function Maze({boxSize, setMazeCompleted, setCurrentMinion, minions, setCurrentT
-            classes: classes[value as value]
-          }
-        }
-        return newMaze;
-      })
-    }
-    else if (allTilesHidden) {
+    async function mazeInit(){
       const mazeTiles = document.getElementsByClassName('mazeTile');
-      const halfway = Math.floor(mazeTiles.length/2);
-      let index = -1;
-      function step() {
-        for (let i = 0; i < 50 - 49*((Math.abs(halfway - index)/halfway)); i++) {
-          index++;
-          if (index === displayVisited.length) break;
-          const currentTile = mazeTiles[displayVisited[index] as number];
-          currentTile.classList.remove('showNone');
-        }
+      if (mazeGenerated === false) {
+        setMazeGenerated(true);
+        // const {graph, visited, classes, towers} = generateMaze(width,height)
+        
+        const {graphBE, visited, classes, towers} = await apiService.createMaze()
+        let graph = new Graph()
+        graph.reAssign(graphBE)
+        console.log({graph, visited, classes, towers});
+        
+        setTowers(() => towers.map((tower:any) => {
+          return {
+            id: tower[0],
+            xPos: tower[0]%width,
+            yPos: Math.floor(tower[0]/width),
+            numbers: tower[1],
+            color: 'red',
+            minion: null,
+            minionAlignment: null,
+            alignment: 'none',
+            animations: [],
+            minionSortingSpeed: null,
+            sortingAlgo: 'bubble'
+          }
+        })
+        )
+        setCurrentGraph(graph);
+        setDisplayVisited(visited);
+        setMaze(oldMaze => {
+          const newMaze = [...oldMaze.maze];
+          for (let value of visited) {
+            newMaze[value as number] = {
+              ...newMaze[value as number],
+              classes: classes[value as value]
+            }
+          }
+          return {...oldMaze, maze: newMaze};
+        })
+        
+      }
+      else if (allTilesHidden && currentGraph) {
+        const mazeTiles = document.getElementsByClassName('mazeTile');
+        console.log({mazeTiles})
+        const halfway = Math.floor(mazeTiles.length/2);
+        let index = -1;
+        function step() {
+          for (let i = 0; i < 50 - 49*((Math.abs(halfway - index)/halfway)); i++) {
+            index++;
+            if (index === displayVisited.length) break;
+            const currentTile = mazeTiles[displayVisited[index] as number];
+            currentTile.classList.remove('showNone');
+          }
 
-        if (index < displayVisited.length) requestAnimationFrame(step)
-        else {
-          setAllTilesHidden(false);
-          setMazeCompleted();
-        };
+          if (index < displayVisited.length) requestAnimationFrame(step)
+          else {
+            setAllTilesHidden(false);
+            setMazeCompleted();
+          };
+        }
+        requestAnimationFrame(step);
       }
-      requestAnimationFrame(step);
     }
+    mazeInit();
   },[maze]);
 
   return (
     <>
```

Comment:
This is already in a div, do we need an empty tag as well?

Reply:
No we don't